Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -126,17 +130,34 @@ def toXML(self, writer, ttFont, progress=None):
 			counter = counter + 1
 			glyph = self[glyphName]
 			if glyph.numberOfContours:
-				writer.begintag('TTGlyph', [
-						("name", glyphName),
-						("xMin", glyph.xMin),
-						("yMin", glyph.yMin),
-						("xMax", glyph.xMax),
-						("yMax", glyph.yMax),
-						])
-				writer.newline()
-				glyph.toXML(writer, ttFont)
-				writer.endtag('TTGlyph')
-				writer.newline()
+				if splitGlyphs:
+					path, ext = os.path.splitext(tablePath)
+					fileNameTemplate = path + ".%s" + ext
+					glyphPath = fileNameTemplate % nameToIdentifier(glyphName)
+					glyphWriter = xmlWriter.XMLWriter(glyphPath, idlefunc=idlefunc,
+				      newlinestr=newlinestr)
+					glyphWriter.begintag("ttFont", ttLibVersion=ttLibVersion)
```

Comment:
I wonder, wouldn't it be better to have "glyf" as the root tag of the single TTGlyph ttx file rather than a generic "ttFont"?
Other tables could receive a similar treatment, and it'd good to keep a reference to the container element in the split file.

Reply:
How strongly do you feel about this suggestion?  My thought is that it is preferable to have every file generated by the ttx tools be normalized with a ttFont root tag with version information so we can verify / act if the file isn't generated by ttx tools or the 'right' version of ttx.  When a font is split between many different files, it is possible for different files to be generated with different versions of the tool (i.e. the files are disconnected and can be read/written separately).  

Additionally, it is also nice to be able to share the existing xmlReader which always expects the ttFont to be the root node with version.  

What would the use case for keeping a reference to the container element be?  The tables which are split do not have a reference back to the containing ttf font b/c they don't have a need for it.  The glyph's split from the glyp table have the similar situation - they don't have any need for information in the containing table.  The pattern which I attempted to follow seems to be that the readers responsibility is to read information and make available to their owner who puts the information in context (i.e. font -> table, glyp->glyph).

I can invest in making the change as you suggest, but it will be a bit larger change and I'm not sure that I understand the motivation.

Reply:
Maybe we could keep both the root `ttFont` element (with the ttLibVersion attribute) and the second-level container element (e.g. `glyf`).
The use case I see is mostly clarity for the reader, to better the hierarchy of the data. The TTGlyph element does not "belong" to ttFont directly, but to the glyf table, which in turns is contained in the ttFont.
If it's too much trouble to you adding this, you can leave it there and I'll pick it up once I find the time.
Thanks for you work!

Reply:
I think that this is good.  I made the necessary changes.  Now both ttFont and glyf are present in the generated glyf file.  I've updated the xmlReader test so that it expects this new format and passes.